content/en/blog/2022/centralized-east-west-gateway/index.md

```diff
@@ -0,0 +1,293 @@
+
+Examples are demonstrated in the next section.
+
+## Related works
+
+There are a few proposals in the istio community to address those problems mentioned above. For example, sidecars can be installed on each node instead of each pod, [1]. This way, the number of sidecars used can be reduced dramatically, and the delay is also reduced because the requests only need to go through one sidecar. However, this approach needs modifications on the node, and it can also cause resource competition between application and sidecars. Moreover, the upgrading and maintenance for the sidecars can be as demanding as the per-pod sidecar mode. 
+
+A recently released document from istio proposes a similar centralized sidecar idea[2]. Since the sidecars are moved out of the nodes, it faces the similar problems we mentioned above: how to accomplish mtls and tracing features that are covered by the local sidecars. The solution used is to install an agent on the node. By this way, it provides comparable features to the original sidecars. However, because an agent is involved, complexity is introduced when the agent is installed or upgraded. It's a great solution for applications where those features are needed. For the HTTP applications that do not require mtls and end-to-end tracing, we still recommend our solution, where no modification on node is required.
+
+## Configuration examples
+
+
+Next, we introduce how to deploy centralized east-west traffic gateway.
+
+Prerequisite:
+
+1. Only one Gateway can be attached to InternalGateway.
+2.  The VirtualService managed by InternalGateway must be attached to the Gateway mention in 1.
+3. CoreDNS is required.
+
+### Use centralized east-west traffic gateway to handle east-west traffic
+Step 1: Deploy InternalGateway service
+
+Step 2: Create HelloWorld service. Note that the port opened must be consistent with the port
+opened on the gateway.
+
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: helloworld
+  labels:
+    app: helloworld
+spec:
+  ports:
+  - port: 5000
+    name: http
+  selector:
+    app: helloworld
+```
+
+Step 3: Create Gateway, and attach it to InternalGateway.
+```yaml
+kind: Gateway
+metadata:
+  name: internalGateway
+  namespace: istio-system
+spec:
+  selector:
+    app: internalGateway
+  servers:
+  - port:
+      number: 5000
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+```
+
+Step 4: Create VirtualService.
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: helloworld
+  namespace: istio-system
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - internalGateway
+  http:
+  - match:
+    - uri:
+        exact: /hello
+    route:
+    - destination:
+        host: helloworld.default.svc.cluster.local
+```
+Step 5: Execute the following command. You should find the IP address of the HelloWorld service changed to the gateway's IP.
```

Comment:
how? Is there some third party code in play here? If so, is it open source? If so, where is the code?

Reply:
Thanks for the attention and comments!

Yes, additional code is needed to accomplish what we proposed here. We introduced a controller named Canal controller. Canal controller watches configuration changes in services and redirects traffic to the internal gateway through core DNS. For data planes, we use router mode Envoys, and envoy’s configuration goes through istiod as usual.

The code is still going through a company procedure before it can be open source. The earliest available date will be 9/23.  I will update this PR then. 

Thanks again!

Reply:
Here is a pointer to the open source code. Thanks for your attention! 

https://github.com/alibaba/alibaba-centralized-mesh-gateway

Reply:
Looks like I am not alone asking this q :) Is it possible to write the blog highlighting what you had to change to build it on top of Istio?